app/uk/gov/hmrc/gform/eval/EvaluationResults.scala

```diff
@@ -83,7 +83,7 @@ case class EvaluationResults(
   // Sum field may be hidden by AddToList or by Revealing choice
   private def isSumHidden(modelComponentId: ModelComponentId): Boolean = {
     val expr = FormCtx(modelComponentId.toFormComponentId)
-    exprMap.get(expr).fold(true)(_ === Hidden)
+    exprMap.get(expr).fold(false)(_ === Hidden)
```

Comment:
this is the change that fixes the issue. @josef-vlach can you please recommend if this doesn't break anything else?

Reply:
It looks like group + sum combination is broken in more than one way. Given template below even `${textA.sum}` expression seems to be broken.

```
    {
      "title": "Page A",
      "fields": [
        {
          "id": "fieldA",
          "type": "text",
          "label": "Field A",
          "format": "sterling"
        }
      ]
    },
    {
      "title": "About page",
      "includeIf": "${fieldA < 0}",
      "fields": [
        {
          "id": "group",
          "type": "group",
          "label": "Group",
          "fields": [
            {
              "id": "textA",
              "type": "text",
              "label": "Text A",
              "format": "sterling"
            },
            {
              "id": "totalTextA",
              "type": "text",
              "label": "Total A",
              "format": "sterling",
              "value": "${fieldA}",
              "submitMode": "readonly",
              "presentationHint": "totalValue"
            }
          ]
        }
      ]
    },
    {
      "title": "Page Last",
      "fields": [
        {
          "id": "lastInfo",
          "type": "info",
          "label": "",
          "infoText": "${textA.sum} :: ${totalTextA.sum}"
        }
      ]
    }
    ```

Reply:
@girishkamat To elaborate, [looking at `DependencyGraph`](http://localhost/submissions/gform-plugin/exprs/total-value-in-group) for the form above reveals that said graph is not correct:

```
"graph": [
  {
    "layerNumber": "0",
    "nodes": "List(Simple(1_group), Simple(1_totalTextA), Simple(lastInfo))"
  },
  {
    "layerNumber": "1",
    "nodes": "List(Expr(Constant(0)), Expr(FormCtx(fieldA)), Expr(FormCtx(textA)), Expr(FormCtx(totalTextA)))"
  },
  {
    "layerNumber": "2",
    "nodes": "List(Simple(fieldA), Simple(textA), Simple(totalTextA))"
  }
]
```

`"layerNumber": "2"` is definitely wrong. `fieldA` and `textA` can't be on the same layer.